kip-0039.md

```diff
@@ -0,0 +1,134 @@
+---
+KIP: '0039'
+Title: kadena_getNetwork_v1 API
+Author: Danillo Felixdaal and Bart Huijgen
+Status: Draft
+Type: Standard
+Category: Interface
+Created: 2025-02-17
+---
+
+# kadena_getNetwork_v1 API
+
+## Abstract
+
+This KIP standardizes the `kadena_getNetwork_v1` method for retrieving information about the currently active network from a Kadena wallet. The method provides consistent network details to dApps, ensuring they can correctly adjust to the appropriate blockchain environment.
+
+## Motivation
+
+A uniform approach to obtaining network information simplifies dApp development by ensuring that network details are provided in a consistent format across different wallet implementations. This standardization will:
+
+- Provide reliable network details (such as network name and networkId) to dApps.
+- Simplify integration and reduce potential mismatches between wallet and dApp network settings.
+- Enhance the overall user experience by ensuring clear network context.
+
+## Specification
+
+### Method Definition
+
+**Method Name:** `kadena_getNetwork_v1`
+
+### Request Parameters
+
+```typescript
+interface GetNetworkRequest {
+  // No parameters are required for this method.
+}
+```
+
+### Response
+
+The method returns a NetworkInfo object containing details about the active network:
+
+```typescript
+interface NetworkInfo {
+  name: string // The human-readable name of the network.
+  networkId: string // The unique identifier for the network (e.g., "mainnet01").
+  url?: string // Optional: The network endpoint URL.
+}
+```
+
+### Error Responses
+
+In case of failure, the wallet MUST return an error according to the [jsonrpc spec](https://www.jsonrpc.org/specification#error_object) which represent the following type:
+
+```typescript
+interface ErrorResponse {
+  id: number
+  jsonrpc: '2.0'
+  error: {
+    code: number
+    message: string
+    data?: object
+  }
+}
+```
+
+### Error types
+
+| Code   | Message               |
+| ------ | --------------------- |
+| -32603 | Internal server error |
+
+### Behavior Requirements
+
+1. Network Information Retrieval:
+
+   - MUST return accurate and up-to-date information about the currently active network.
+   - MUST support all networks that the wallet is configured to operate with.
+
+2. Error Handling:
+   - MUST return a clear error if network information cannot be retrieved due to configuration issues or unsupported networks.
+
+# Examples
+
+## Basic Get Network Request
+
+### Request
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "kadena_getNetwork_v1",
+  "params": {}
+}
+```
+
+### Success Response
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "result": {
+    "name": "kadena:mainnet01",
```

Comment:
I wouldn't use `kadena:` in this. The RPC will never return `ethereum:xyz` so it's kinda redundant.
I would consider `networkName` as the display name to show the user in the UI

```suggestion
    "name": "Mainnet",
```